lib/cc/engine/analyzers/command_line_runner.rb

```diff
@@ -14,14 +15,33 @@ def initialize(command, timeout = DEFAULT_TIMEOUT)
 
         def run(input)
           Timeout.timeout(timeout) do
-            out, err, status = Open3.capture3(command, stdin_data: input)
+            file_actions = Spoon::FileActions.new
+            id = SecureRandom.hex
 
-            status ||= handle_open3_race_condition(out)
+            # setup stdin
+            file_actions.close(0)
+            File.open("/tmp/#{id}.in", "w") { |f| f.write(input) }
+            file_actions.open(0, "/tmp/#{id}.in", 0, 0600)
 
-            if status.success?
-              yield out
+            # setup stdout
+            file_actions.close(1)
+            file_actions.open(1, "/tmp/#{id}.out", File::WRONLY | File::TRUNC | File::CREAT, 0600)
+
+            # setup stderr
+            file_actions.close(2)
+            file_actions.open(2, "/tmp/#{id}.err", File::WRONLY | File::TRUNC | File::CREAT, 0600)
```

Comment:
FWIW these options are the ones I'm least confident about. They're mostly copied from [this example](https://github.com/headius/spoon/blob/5b83982bf617834497d646fc21768e281ad1deda/examples/ls.rb).

The expression `File::WRONLY | File::TRUNC | File::CREAT` evaluates to `1537`. I'm not sure if `TRUNC` means that stderr and stdout will be truncated, but that wouldn't be great.